components/motor/gpio/motor_encoder.go

```diff
@@ -379,7 +387,18 @@ func (m *EncodedMotor) setPower(ctx context.Context, powerPct float64, internal
 func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extra map[string]interface{}) error {
 	ctx, done := m.opMgr.New(ctx)
 	defer done()
-	if err := m.goForInternal(ctx, rpm, revolutions); err != nil {
+
+	direction := sign(rpm * revolutions)
+
+	currentPos, err := m.position(ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	goalPos := (math.Abs(revolutions) * m.ticksPerRotation * direction) + currentPos
+	goalRPM := math.Abs(rpm) * m.state.direction
```

Comment:
what's the motivation for moving these from goForInternal to GoFor? I don't have a strong opinion about it, just curious

Reply:
jk I see to remove old line 392 so there's less state access